panel/models/custom_html.ts

```diff
@@ -0,0 +1,155 @@
+import * as p from "@bokehjs/core/properties"
+import {ModelEvent, JSON} from "@bokehjs/core/bokeh_events"
+import {HTMLBox, HTMLBoxView} from "@bokehjs/models/layouts/html_box"
+
+import {CachedVariadicBox, set_size} from "./layout"
+
+class DOMEvent extends ModelEvent {
+  event_name: string = "dom_event"
+
+  constructor(readonly element: string, readonly event: any) {
+    super()
+  }
+
+  protected _to_json(): JSON {
+    return {model: this.origin, element: this.element, event: this.event}
+  }
+}
+
+function simplify(event: any): any {
+  const copy: any = {}
+  for (const property in event) {
+    const ptype = typeof event[property] 
+    if(ptype === "function" || ptype === 'object')
+      continue
+    copy[property] = event[property];    
+  }
+  return copy
+}
+
+function isNumeric(str: any): any {
+  if (typeof str != "string")
+    return false  
+  return !isNaN(str) && !isNaN(parseFloat(str))
+}
+
+export class CustomHTMLView extends HTMLBoxView {
+  model: CustomHTML
+  _prev_sizing_mode: string | null
+  _changing: boolean = false
+  protected divEl: HTMLElement
+
+  connect_signals(): void {
+    super.connect_signals()
+    const resize = () => {
+      this.render()
+      this.root.compute_layout() // XXX: invalidate_layout?
+    }
+
+    this.connect(this.model.properties.height.change, resize)
+    this.connect(this.model.properties.width.change, resize)
+    this.connect(this.model.properties.height_policy.change, resize)
+    this.connect(this.model.properties.width_policy.change, resize)
+    this.connect(this.model.properties.sizing_mode.change, resize)
+    this.connect(this.model.properties.html.change, () => this.render())
+	this.connect(this.model.model.change, () => {
+	  if (!this._changing)
+	    this.render()
+	})
+  }
+
+  private _render_html(literal: string, params: any): string {
+    return new Function("ns", "with (ns) { return `"+literal+"`; }")(params)
+  }
+
+  render(): void {
+    super.render()
+    this.divEl = document.createElement('div')
+    this.divEl.innerHTML = this._render_html(this.model.html, this.model.model)
+
+    set_size(this.divEl, this.model)
+    this.el.appendChild(this.divEl)
+
+    const id = this.model.model.id
+	for (const name in this.model.attrs) {
+	  const el: any = document.getElementById(`${name}-${id}`)
+      if (el == null) {
+        console.warn(`DOM node '${name}-${id}' could not be found.`)
+        continue
+      }
+	  const observer = new MutationObserver(() => {
+		this._update_model(el, name)
+	  })
+	  observer.observe(el, {characterData: true, attributes: true, childList: true, subtree: true});
```

Comment:
If this is supposed to be used with complex element trees, then I would recommend benchmarking this in common browsers, because last time I tried to use it, it resulted in abysmal performance.

Reply:
Wouldn't expect these components to be deeply nested but I would like to find a way to disable or configure this or indeed limit it to attribute changes.